src/source/Ice/IceAgent.c

```diff
@@ -705,6 +706,10 @@ STATUS iceAgentPopulateSdpMediaDescriptionCandidates(PIceAgent pIceAgent, PSdpMe
     while (pCurNode != NULL) {
         CHK_STATUS(doubleListGetNodeData(pCurNode, &data));
         pCurNode = pCurNode->pNext;
+        pCandidate = (PIceCandidate) data;
+        if (pCandidate->state != ICE_CANDIDATE_STATE_VALID) {
```

Comment:
Consider not using flow control here - we try to avoid continue and break in the condebase.

Instead, think if it's better 

if (pCandidate->state == ICE_CANDIDATE_STATE_VALID) {
 STRCPY(pSdpMediaDescription->sdpAttributes[attrIndex].attributeName, "candidate");
        CHK_STATUS(iceCandidateSerialize((PIceCandidate) data, pSdpMediaDescription->sdpAttributes[attrIndex].attributeValue, &attrBufferLen));
        attrIndex++;
}